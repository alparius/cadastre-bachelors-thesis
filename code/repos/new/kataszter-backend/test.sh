#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
