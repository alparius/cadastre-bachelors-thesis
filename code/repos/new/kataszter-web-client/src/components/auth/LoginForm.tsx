import React, { useEffect, useState } from "react";
import { FormattedMessage as FM, useIntl } from "react-intl";
import { NavLink } from "react-router-dom";
import { Button, Form, Grid, Header, Message } from "semantic-ui-react";

import ILoginUserData from "../../data/user/LoginUserData";
import ILabelResponse from "../../requests/response/LabelResponse";
import WrappedRecaptcha from "./WrappedRecaptcha";

interface ILoginFormProps {
    loginUserData: ILoginUserData;
    response: ILabelResponse | undefined;
    recaptchaToken: string;
    postLogin(recaptchaToken: string): Promise<void>;
    setLoginUserData(newLoginUserData: ILoginUserData): void;
    setRecaptchaToken(recaptchaToken: string): void;
}

const LoginForm: React.FC<ILoginFormProps> = (props: ILoginFormProps) => {
    const { loginUserData, response, recaptchaToken, postLogin, setLoginUserData, setRecaptchaToken } = props;
    const { formatMessage: fm } = useIntl();

    const [errorMsgs, setErrorMsgs] = useState(new Array<string>());
    const [fieldErrors, setFieldErrors] = useState(new Array<string>());
    const [recaptchDOMKey, setRecaptchaDOMKey] = useState(0);

    useEffect(() => {
        if (response && response.label.startsWith("error")) {
            setRecaptchaDOMKey((prevCount) => prevCount + 1);
        }
    }, [response, setRecaptchaToken]);

    const handleRecaptcha = (value: string) => {
        setRecaptchaToken(value);
    };

    const handleChange = (_: any, { name, value }: any) => {
        setLoginUserData({ ...loginUserData, [name]: value });
    };

    const handleLoginButton = () => {
        setErrorMsgs(new Array<string>());
        setFieldErrors(new Array<string>());

        const tobeFieldErrors = new Array<string>();
        const tobeErrorMsgs = new Array<string>();

        Object.entries({ ...loginUserData, recaptchaToken: "placeholder" }).forEach((field: [string, string]) => {
            if (field[1] === "") {
                tobeFieldErrors.push(field[0]);
            }
        });
        if (tobeFieldErrors.length !== 0) {
            tobeErrorMsgs.push("formError.emptyFields");
        }

        if (!loginUserData.email.includes("@")) {
            tobeErrorMsgs.push("formError.invalidEmail");
            tobeFieldErrors.push("email");
        }

        if (recaptchaToken === "") {
            tobeErrorMsgs.push("error.recaptcha");
            tobeFieldErrors.push("recaptcha");
        }

        if (tobeFieldErrors.length === 0) {
            postLogin(recaptchaToken);
            setErrorMsgs(new Array<string>());
            setFieldErrors(new Array<string>());
        } else {
            setFieldErrors(tobeFieldErrors);
            setErrorMsgs(tobeErrorMsgs);
        }
    };

    return (
        <Grid textAlign="center">
            <Grid.Row>
                <Grid.Column style={{ maxWidth: 450 }}>
                    <Form size="large">
                        {response && (
                            <Message negative>
                                <FM id={response.label} />
                            </Message>
                        )}
                        {errorMsgs.length !== 0 && (
                            <Message negative>
                                {errorMsgs.map((msgID: string) => (
                                    <React.Fragment key={msgID}>
                                        <FM id={msgID} />
                                        <br />
                                    </React.Fragment>
                                ))}
                            </Message>
                        )}
                        <Form.Input
                            error={fieldErrors.includes("email")}
                            type="text"
                            name="email"
                            placeholder={fm({ id: "auth.email" })}
                            value={loginUserData.email}
                            onChange={handleChange}
                            fluid
                            icon="user"
                            iconPosition="left"
                        />
                        <Form.Input
                            error={fieldErrors.includes("password")}
                            type="password"
                            name="password"
                            placeholder={fm({ id: "auth.password" })}
                            value={loginUserData.password}
                            onChange={handleChange}
                            fluid
                            icon="lock"
                            iconPosition="left"
                        />
                        <Header as="h5" style={{ marginTop: "0em" }}>
                            <NavLink to="/forgotPassword" className="hoverLink">
                                <FM id="auth.loginpage.forgotPassword" />
                            </NavLink>
                        </Header>

                        <WrappedRecaptcha domkey={recaptchDOMKey} handleRecaptcha={handleRecaptcha} />

                        <Button color="violet" fluid size="large" onClick={handleLoginButton}>
                            <FM id="auth.login" />
                        </Button>
                    </Form>
                </Grid.Column>
            </Grid.Row>

            <Grid.Row style={{ marginTop: "10vh" }}>
                <Grid.Column style={{ maxWidth: 800 }}>
                    <Header as="h3">
                        <FM id="auth.loginpage.registerCall" />{" "}
                        <NavLink to="/register" className="hoverLink">
                            <FM id="auth.loginpage.register" />
                        </NavLink>
                    </Header>
                    <Header as="h4">
                        <FM id="auth.loginpage.registerNote" />
                    </Header>
                </Grid.Column>
            </Grid.Row>
        </Grid>
    );
};

export default LoginForm;
