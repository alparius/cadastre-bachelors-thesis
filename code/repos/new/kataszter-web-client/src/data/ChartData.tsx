interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
