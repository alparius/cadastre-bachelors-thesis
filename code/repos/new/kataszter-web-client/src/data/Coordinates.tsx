interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
