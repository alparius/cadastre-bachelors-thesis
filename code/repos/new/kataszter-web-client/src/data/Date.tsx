interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
