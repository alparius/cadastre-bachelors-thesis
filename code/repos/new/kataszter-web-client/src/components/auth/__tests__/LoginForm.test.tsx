/*===================
  LoginForm.unit.test.tsx
 ===================*/
import React from "react";

import { NewTestLabelResponseError, NewTestLoginUserData } from "../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter, mountWithIntlAndRouter } from "../../../util/test/test-wrapper";
import LoginForm from "../LoginForm";

const handlers = {
    setLoginUserData: jest.fn(),
    postLogin: jest.fn(),
    setRecaptchaToken: jest.fn()
};

describe("LoginForm element", () => {
    it("should render the login form with input fields", () => {
        const container = createComponentWithIntlAndRouter(
            <LoginForm loginUserData={NewTestLoginUserData} recaptchaToken="" response={undefined} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a message too when there is a response", () => {
        const container = createComponentWithIntlAndRouter(
            <LoginForm loginUserData={NewTestLoginUserData} recaptchaToken="" response={NewTestLabelResponseError} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call the setLoginUserData function when the form gets input", () => {
        const container = mountWithIntlAndRouter(
            <LoginForm loginUserData={NewTestLoginUserData} recaptchaToken="" response={undefined} {...handlers} />
        );
        let input = container.find('input[name="email"]');
        input.simulate("change", { target: { value: "test" } });
        input = container.find('input[name="password"]');
        input.simulate("change", { target: { value: "test" } });
        expect(handlers.setLoginUserData).toHaveBeenCalledTimes(2);
    });

    it("should not call the postLogin function when the login gets clicked but the recaptchaToken is not valid", () => {
        const container = mountWithIntlAndRouter(
            <LoginForm loginUserData={NewTestLoginUserData} recaptchaToken="" response={undefined} {...handlers} />
        );
        const input = container.find("button");
        input.simulate("click");
        expect(handlers.postLogin).toHaveBeenCalledTimes(0);
    });
});
