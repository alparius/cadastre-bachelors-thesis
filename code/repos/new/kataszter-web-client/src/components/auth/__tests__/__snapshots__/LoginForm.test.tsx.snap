// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm element should render a message too when there is a response 1`] = `
<div
  className="ui center aligned grid"
>
  <div
    className="row"
  >
    <div
      className="column"
      style={
        Object {
          "maxWidth": 450,
        }
      }
    >
      <form
        className="ui large form"
        onSubmit={[Function]}
      >
        <div
          className="ui negative message"
        >
          error.test
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
          >
            <input
              name="email"
              onChange={[Function]}
              placeholder="E-mail address"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="user icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
          >
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <i
              aria-hidden="true"
              className="lock icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <h5
          className="ui header"
          style={
            Object {
              "marginTop": "0em",
            }
          }
        >
          <a
            aria-current={null}
            className="hoverLink"
            href="/forgotPassword"
            onClick={[Function]}
          >
            Forgotten Password?
          </a>
        </h5>
        <div
          style={
            Object {
              "marginBottom": "3em",
              "marginTop": "1em",
            }
          }
        >
          <div
            className="g-recaptcha"
            data-badge="bottomright"
            data-sitekey="6LfURxEbAAAAAI6TU9GsuydKqmXb3t2TDZwvOYd2"
            data-size="normal"
            data-tabindex="0"
            data-theme="light"
            data-type="image"
            id="g-recaptcha"
          />
        </div>
        <button
          className="ui violet large fluid button"
          onClick={[Function]}
        >
          Log in!
        </button>
      </form>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "marginTop": "10vh",
      }
    }
  >
    <div
      className="column"
      style={
        Object {
          "maxWidth": 800,
        }
      }
    >
      <h3
        className="ui header"
      >
        No account yet?
         
        <a
          aria-current={null}
          className="hoverLink"
          href="/register"
          onClick={[Function]}
        >
          Register!*
        </a>
      </h3>
      <h4
        className="ui header"
      >
        *Note that the site is closed to the public, new registrations need to be approved by an admin.
      </h4>
    </div>
  </div>
</div>
`;

exports[`LoginForm element should render the login form with input fields 1`] = `
<div
  className="ui center aligned grid"
>
  <div
    className="row"
  >
    <div
      className="column"
      style={
        Object {
          "maxWidth": 450,
        }
      }
    >
      <form
        className="ui large form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
          >
            <input
              name="email"
              onChange={[Function]}
              placeholder="E-mail address"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="user icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
          >
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <i
              aria-hidden="true"
              className="lock icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <h5
          className="ui header"
          style={
            Object {
              "marginTop": "0em",
            }
          }
        >
          <a
            aria-current={null}
            className="hoverLink"
            href="/forgotPassword"
            onClick={[Function]}
          >
            Forgotten Password?
          </a>
        </h5>
        <div
          style={
            Object {
              "marginBottom": "3em",
              "marginTop": "1em",
            }
          }
        >
          <div
            className="g-recaptcha"
            data-badge="bottomright"
            data-sitekey="6LfURxEbAAAAAI6TU9GsuydKqmXb3t2TDZwvOYd2"
            data-size="normal"
            data-tabindex="0"
            data-theme="light"
            data-type="image"
            id="g-recaptcha"
          />
        </div>
        <button
          className="ui violet large fluid button"
          onClick={[Function]}
        >
          Log in!
        </button>
      </form>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "marginTop": "10vh",
      }
    }
  >
    <div
      className="column"
      style={
        Object {
          "maxWidth": 800,
        }
      }
    >
      <h3
        className="ui header"
      >
        No account yet?
         
        <a
          aria-current={null}
          className="hoverLink"
          href="/register"
          onClick={[Function]}
        >
          Register!*
        </a>
      </h3>
      <h4
        className="ui header"
      >
        *Note that the site is closed to the public, new registrations need to be approved by an admin.
      </h4>
    </div>
  </div>
</div>
`;
